ctx.go

```diff
@@ -258,23 +293,16 @@ var ctxPool = sync.Pool{
 	New: func() interface{} { return &Context{} },
 }
 
-func getCtx(rw http.ResponseWriter, req *http.Request, p router.Params) *Context {
+func getCtx(rw http.ResponseWriter, req *http.Request, p router.Params, s *Server) *Context {
 	ctx, ok := ctxPool.Get().(*Context)
 	if !ok {
 		ctx = &Context{}
 	}
-	ctx.ResponseWriter, ctx.Req, ctx.Params = rw, req, p
+	ctx.ResponseWriter, ctx.Req, ctx.Params, ctx.s = rw, req, p, s
```

Comment:
Any reason this can't be done as three lines. I feel like it's kind of hard to read for maintainers.